six/modules/c++/six.sicd/tests/test_compare_sicd_meshes.cpp

```diff
@@ -0,0 +1,363 @@
+                       "SICD",
+                       1 ,
+                       1);
+
+    parser.addArgument("testSicdPathname",
+                       "Specify a test SICD to compare against the baseline SICD",
+                       cli::STORE,
+                       "testSicdPathname",
+                       "SICD",
+                       1 ,
+                       1);
+
+    return std::auto_ptr<cli::Results>(parser.parse(argc, argv));
+}
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        std::auto_ptr<cli::Results> options = parseCommandLine(argc, argv);
+        const std::string baselineSicdPathname =
+                options->get<std::string>("baselineSicdPathname");
+
+        const std::string testSicdPathname =
+                options->get<std::string>("testSicdPathname");
+
+        std::auto_ptr<six::sicd::NoiseMesh> baselineNoiseMesh;
+        std::auto_ptr<six::sicd::ScalarMesh> baselineScalarMesh;
+        readMeshes(baselineSicdPathname,
+                   baselineNoiseMesh,
+                   baselineScalarMesh);
+
+        std::auto_ptr<six::sicd::NoiseMesh> testNoiseMesh;
+        std::auto_ptr<six::sicd::ScalarMesh> testScalarMesh;
+        readMeshes(testSicdPathname,
+                   testNoiseMesh,
+                   testScalarMesh);
+
+        std::cout << "Comparing Noise Meshes..." << std::endl;
+        if (!compareNoiseMeshes(baselineSicdPathname,
+                                *baselineNoiseMesh,
+                                testSicdPathname,
+                                *testNoiseMesh))
+        {
+            return 1;
+        }
+
+        if (baselineScalarMesh.get() && testScalarMesh.get())
+        {
+            std::cout << "Comparing Scalar Meshes..." << std::endl;
+            if (!compareScalarMeshes(baselineSicdPathname,
+                                     *baselineScalarMesh,
+                                     testSicdPathname,
+                                     *testScalarMesh))
+            {
+                return 1;
+            }
+        }
+        else if (baselineScalarMesh.get() || testScalarMesh.get())
+        {
+            const std::string missingScalarMeshPathname =
+                    baselineScalarMesh.get() ?
+                            testSicdPathname : baselineSicdPathname;
+
+            std::cerr << "Failure - scalar mesh not found in "
+                      << missingScalarMeshPathname << std::endl;
+            return 1;
+        }
+
+        std::cout << "Success - mesh data matches." << std::endl;
+        return 0;
+
+    }
+    catch (const except::Exception& e)
+    {
+        std::cerr << e.getMessage() << std::endl;
+        return 1;
+    }
+    catch (const std::exception& e)
+    {
+        std::cerr << e.what() << std::endl;
+        return 1;
+    }
+    catch (...)
+    {
+        std::cerr << "Unknown exception" << std::endl;
+        return 1;
+    }
+
+    return 0;
```

Comment:
Remove the returns inside each catch block and just have a single `return 1` here